ui/app/src/components/ViewVersionDialog/ContentFieldView.tsx

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (C) 2007-2024 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ContentTypeField } from '../../models';
+import Typography from '@mui/material/Typography';
+import { MonacoWrapper } from '../MonacoWrapper';
+import Box from '@mui/material/Box';
+import AsyncVideoPlayer from '../AsyncVideoPlayer';
+import { asLocalizedDateTime, convertUtcTimeToTimezone } from '../../utils/datetime';
+import Tooltip from '@mui/material/Tooltip';
+import { FormattedMessage } from 'react-intl';
+import Button from '@mui/material/Button';
+import React, { useState } from 'react';
+import useItemsByPath from '../../hooks/useItemsByPath';
+import useLocale from '../../hooks/useLocale';
+import FieldVersionToolbar from '../CompareVersionsDialog/FieldVersionToolbar';
+import { fromString, serialize } from '../../utils/xml';
+import useMonacoOptions from '../../hooks/useMonacoOptions';
+import { nnou } from '../../utils/object';
+import DiffCollectionItem from '../CompareVersionsDialog/FieldsTypesDiffViews/DiffCollectionItem';
+import ContentInstance from '../../models/ContentInstance';
+import useSelection from '../../hooks/useSelection';
+import { countLines } from '../../utils/string';
+import { useVersionsDialogContext } from '../CompareVersionsDialog';
+
+interface ContentFieldViewProps {
+  content: any; // TODO: add typing
+  field: ContentTypeField;
+  xml?: string;
+  showToolbar?: boolean;
+  contentTypeFields: ContentTypeField[];
+  showToolbarFieldNavigation?: boolean;
+  dynamicHeight?: boolean;
+  onSelectField?(field: ContentTypeField): void;
+}
+
+const typesRenderMap = {
+  text: MonacoWrapper,
+  textarea: MonacoWrapper,
+  html: MonacoWrapper,
+  'node-selector': DefaultView,
+  'checkbox-group': DefaultView,
+  repeat: DefaultView,
+  image: DefaultView,
+  'video-picker': DefaultView,
+  time: DefaultView,
+  'date-time': DefaultView,
+  boolean: DefaultView,
+  'numeric-input': DefaultView,
+  dropdown: DefaultView
+};
+
+interface DefaultViewProps {
+  field: ContentTypeField;
+  contentA: any; // TODO: Add typing
+  xml: string;
+}
+
+function DefaultView(props: DefaultViewProps) {
```

Comment:
Move to its own file. There's another component with a similar name; this one might be worth a rename to something more specific that distinguishes it and makes it clear what this one is against the other